src/main/java/de/qabel/desktop/daemon/sync/worker/index/sqlite/SqliteSyncIndex.java

```diff
@@ -0,0 +1,75 @@
+package de.qabel.desktop.daemon.sync.worker.index.sqlite;
+
+import de.qabel.desktop.daemon.sync.worker.index.SyncIndex;
+import de.qabel.desktop.daemon.sync.worker.index.SyncIndexEntry;
+import de.qabel.desktop.daemon.sync.worker.index.SyncState;
+import de.qabel.desktop.nio.boxfs.BoxFileSystem;
+import de.qabel.desktop.nio.boxfs.BoxPath;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+public class SqliteSyncIndex implements SyncIndex {
```

Comment:
It doesn't just fit. Because the sync index exists per sync, it uses another sqlite-file=another connection, is initialized later (after everything else =identities,accounts&syncy has been migrated and loaded) and maybe even later when a sync is recreated that existed previously. So it somehow doesn't fit for the DesktopClientDatabase.
One could consider extracting an SqliteSyncIndexRepository that hooks on every syncIndex and saves it as soon as the syncedState is changed but as the SyncIndex needs to check the repo for uncached entries, they would be tightly coupled. I'm not sure if we win much and we couldn't use the DesktopClientDatabase anyways.

But I'll look into this if we can leverage the existing migrations